File: 数据结构和算法/python/insertionSort.py
# 插入算法及其实现

def insertionSort(array):
    i = 1
    while i < len(array):
        value = array[i]
        j = i - 1
        while j >= 0:
            if array[j] > value:
                array[j+1] = array[j]
            else:
                break
            j -= 1
        array[j+1] = value
        i += 1
    return array

if __name__ == "__main__":
    sortArray = insertionSort([1,5,3,7,6,-2,4,9,8,9])
    print (sortArray)

File: 数据结构和算法/java/InsertionSort.java
/**
 * 插入排序的实现
 */
public class InsertionSort{

    public int[] sort(int[] array){
        for (int i = 1; i < array.length; i++){
            int value = array[i];
            int j = i - 1;
            for (; j >= 0; --j){
                if (array[j] > value){
                    array[j + 1] = array[j];
                }
                else{
                    break;
                }
            }
            array[j + 1] = value;
        }
        return array;
    }
    

    public static void main(String[] args) {
        InsertionSort InsertionSort = new InsertionSort();
        int[] array = new int[]{3, 7, 4, 3, -1, 2, 9, 4, 0};
        int[] sortArray = InsertionSort.sort(array);
        for (int a : sortArray){
            System.out.print(a + ", ");
        }
    }

}